newa/cli.py

```diff
@@ -93,6 +111,27 @@ def main(click_context: click.Context, state_dir: str, conf_file: str, debug: bo
         ctx.logger.debug(f'State directory {ctx.state_dirpath} does not exist, creating...')
         ctx.state_dirpath.mkdir(parents=True)
 
+    def _split(s: str) -> tuple[str, str]:
+        """ split key='some value' into a tuple (key, value) """
+        s = s.strip()
+        if ('=' not in s) or s.startswith('='):
+            raise Exception(
+                f'Option --environment {s} has invalid value, key=value format expected!')
+        (k, v) = map(str.strip, s.split('=', 1))
+        # cut-off surrounding " or '
+        if (v.startswith('"') and v.endswith('"')) or (v.startswith("'") and v.endswith("'")):
+            v = v[1:-1]
+        return (k, v)
+
+    # store environment variables and context provided on a cmdline
+    for s in envvars:
+        k, v = _split(s)
+        ctx.cli_environment[k] = v
```

Comment:
Maybe a bit a streamlining? `dict()` accepts a list of key/value tuples, `dict([(key1, value1), (key2, value2), ...])`, exactly what `_split()` returns, therefore something like the following would pack it into one line:

```python
ctx.cli_environment.update(dict(_split(s) for s in envvars))
```

Reply:
fixed